### PR TITLE
sdl2_sound 2.0.1 (new formula)

### PR DESCRIPTION
**Pull request by pthariensflame**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

**Comment by pthariensflame** [comment]
I need some help with the audit issue.  The test actually needs the "literal interpolation" there because otherwise there's the wrong amount of indentation in the string.

**Comment by chenrui333** [comment]
there is some test failure.

**Comment by pthariensflame** [comment]
The audit failure seems unnecessary since it’s a very well-known project.  As for the linkage error, it’s weird and it didn’t happen locally.

File: Formula/sdl2_sound.rb [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      "-DSDLSOUND_DECODER_AIFF=TRUE",
+      "-DSDLSOUND_DECODER_AU=TRUE",
+      "-DSDLSOUND_DECODER_VOC=TRUE",
+      "-DSDLSOUND_DECODER_FLAC=TRUE",
+      "-DSDLSOUND_DECODER_VORBIS=TRUE",
+      "-DSDLSOUND_DECODER_RAW=TRUE",
+      "-DSDLSOUND_DECODER_SHN=TRUE",
+      "-DSDLSOUND_DECODER_MODPLUG=TRUE",
+      "-DSDLSOUND_DECODER_MP3=TRUE",
+      "-DSDLSOUND_DECODER_MIDI=TRUE"
```
Comment (carlocab) [inline Formula/sdl2_sound.rb:33]:
```suggestion
      "-DSDLSOUND_DECODER_MIDI=TRUE",
```

File: Formula/sdl2_sound.rb [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+         File extension "FLAC"
+         File extension "FLA"
+         Written by Ryan C. Gordon <icculus@icculus.org>.
+         https://icculus.org/SDL_sound/
+    EOS
+    if OS.mac?
+      space = " "
+      expected += <<~EOS
+
+        #{space}* Decode audio through Core Audio through
```
Comment (carlocab) [inline Formula/sdl2_sound.rb:123]:
```suggestion
      expected += <<~EOS

        \ * Decode audio through Core Audio through
```

File: Formula/sdl2_sound.rb [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+  homepage "https://icculus.org/SDL_sound/"
+  license "Zlib"
+  head "https://github.com/icculus/SDL_sound.git", branch: "main"
+
+  stable do
+    # Includes fixes for CMake, just ahead of the release tag
+    url "https://github.com/icculus/SDL_sound/archive/06c8946983ca9b9ed084648f417f60f21c0697f1.tar.gz"
+    sha256 "41f4d779192dea82086c8da8b8cbd47ba99b52cd45fdf39c96b63f75f51293e1"
+    version "2.0.1"
+  end
```
Comment (carlocab) [inline Formula/sdl2_sound.rb:12]:
```suggestion
  homepage "https://icculus.org/SDL_sound/"
  # Includes fixes for CMake, just ahead of the release tag
  url "https://github.com/icculus/SDL_sound/archive/06c8946983ca9b9ed084648f417f60f21c0697f1.tar.gz"
  version "2.0.1"
  sha256 "41f4d779192dea82086c8da8b8cbd47ba99b52cd45fdf39c96b63f75f51293e1"
  license "Zlib"
  head "https://github.com/icculus/SDL_sound.git", branch: "main"
```

File: Formula/sdl2_sound.rb [hunk 4]
```diff
@@ -0,0 +1,153 @@
+class Sdl2Sound < Formula
```
Comment (carlocab) [inline Formula/sdl2_sound.rb:1]: Should this be called `sdl2_sound` given that the project is called `SDL_sound`?

Comment (pthariensflame) [reply Formula/sdl2_sound.rb:1]: We already have an `sdl_sound` (for SDL1), and this follows the existing conventions from Homebrew and elsewhere.  (See `sdl2_image` for example.)

File: Formula/sdl2_sound.rb [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+      "-DSDLSOUND_DECODER_VOC=TRUE",
+      "-DSDLSOUND_DECODER_FLAC=TRUE",
+      "-DSDLSOUND_DECODER_VORBIS=TRUE",
+      "-DSDLSOUND_DECODER_RAW=TRUE",
+      "-DSDLSOUND_DECODER_SHN=TRUE",
+      "-DSDLSOUND_DECODER_MODPLUG=TRUE",
+      "-DSDLSOUND_DECODER_MP3=TRUE",
+      "-DSDLSOUND_DECODER_MIDI=TRUE",
+    ]
+    args += ["-D", "SDLSOUND_DECODER_COREAUDIO=TRUE"] if OS.mac?
```
Comment (pthariensflame) [inline Formula/sdl2_sound.rb:32]:
```suggestion
    args += "-DSDLSOUND_DECODER_COREAUDIO=TRUE" if OS.mac?
```

File: Formula/sdl2_sound.rb [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+      "-DSDLSOUND_DECODER_VOC=TRUE",
+      "-DSDLSOUND_DECODER_FLAC=TRUE",
+      "-DSDLSOUND_DECODER_VORBIS=TRUE",
+      "-DSDLSOUND_DECODER_RAW=TRUE",
+      "-DSDLSOUND_DECODER_SHN=TRUE",
+      "-DSDLSOUND_DECODER_MODPLUG=TRUE",
+      "-DSDLSOUND_DECODER_MP3=TRUE",
+      "-DSDLSOUND_DECODER_MIDI=TRUE",
+    ]
+    args += "-DSDLSOUND_DECODER_COREAUDIO=TRUE" if OS.mac?
```
Comment (carlocab) [inline Formula/sdl2_sound.rb:32]:
```suggestion
    args << "-DSDLSOUND_DECODER_COREAUDIO=TRUE" if OS.mac?
```
`+=` only works for arrays here.

**Comment by pthariensflame** [comment]
The rpath issue on Arm seems to still be present.

**Comment by pthariensflame** [comment]
The rpath issue on Arm seems to still be present.

**Comment by cho-m** [comment]
May be due to https://github.com/icculus/SDL_sound/blob/v2.0.1/CMakeLists.txt#L18-L19
```cmake
    # Don't use -rpath.
    set(CMAKE_SKIP_RPATH ON CACHE BOOL "Skip RPATH" FORCE)
```

**Comment by pthariensflame** [comment]
> May be due to https://github.com/icculus/SDL_sound/blob/v2.0.1/CMakeLists.txt#L18-L19
> 
> ```cmake
>     # Don't use -rpath.
>     set(CMAKE_SKIP_RPATH ON CACHE BOOL "Skip RPATH" FORCE)
> ```

Seems likely; what should be done?

File: Formula/sdl2_sound.rb [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+  depends_on "cmake" => :build
+  depends_on "libmodplug"
+  depends_on "sdl2"
+  depends_on "timidity"
+
+  def install
+    args = std_cmake_args
+    args += [
+      "-DCMAKE_INSTALL_RPATH=#{rpath}",
+      "-DSDLSOUND_DECODER_WAV=TRUE",
```
Comment (carlocab) [inline Formula/sdl2_sound.rb:20]:
```suggestion
      "-DCMAKE_INSTALL_RPATH=#{rpath}",
      "-DCMAKE_EXE_LINKER_FLAGS=-Wl,-rpath,#{rpath}",
      "-DSDLSOUND_DECODER_WAV=TRUE",
```

File: Formula/sdl2_sound.rb [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+           File extension "3gp2"
+           File extension "amrf"
+           File extension "amr"
+           File extension "ima4"
+           File extension "ima"
+           Written by Eric Wing <ewing . public @ playcontrol.net>.
+           https://playcontrol.net
+      EOS
+    end
+    assert_equal expected.strip, shell_output("#{bin/"playsound"} --decoders").strip
```
Comment (carlocab) [inline Formula/sdl2_sound.rb:147]:
```suggestion
    assert_equal expected.strip, shell_output("#{bin}/playsound --decoders").strip
```

**Review by carlocab — DISMISSED** [review]

**Comment by yurikoles** [comment]
@pthariensflame could you please add tests form #98073? Just add `pkgshare.install "examples"` at the end of `install` and concatenate our `test` blocks.

File: Formula/sdl2_sound.rb [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+      "-DSDLSOUND_DECODER_AIFF=TRUE",
+      "-DSDLSOUND_DECODER_AU=TRUE",
+      "-DSDLSOUND_DECODER_VOC=TRUE",
+      "-DSDLSOUND_DECODER_FLAC=TRUE",
+      "-DSDLSOUND_DECODER_VORBIS=TRUE",
+      "-DSDLSOUND_DECODER_RAW=TRUE",
+      "-DSDLSOUND_DECODER_SHN=TRUE",
+      "-DSDLSOUND_DECODER_MODPLUG=TRUE",
+      "-DSDLSOUND_DECODER_MP3=TRUE",
+      "-DSDLSOUND_DECODER_MIDI=TRUE",
```
Comment (yurikoles) [inline Formula/sdl2_sound.rb:31]: Those all, except MIDI, are [enabled by default](https://github.com/icculus/SDL_sound/blob/v2.0.1/CMakeLists.txt#L60).

**Comment by pthariensflame** [comment]
Should be done now!

File: Formula/sdl2_sound.rb [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+      -I#{include}/SDL2
+      -I#{Formula["sdl2"].include}/SDL2
+      -L#{lib}
+      -L#{Formula["sdl2"].lib}
+      -lSDL2_sound
+      -lSDL2
+    ]
+    Dir["#{pkgshare}/examples/*.c"].each do |r|
+      system ENV.cc, r, "-o", "#{File.basename(r, ".c")}.out", *flags
+    end
```
[the remark this answers is not in the document]
Comment (yurikoles) [reply Formula/sdl2_sound.rb:151]: They actually work, but launching them without arguments or with `--help` or `--credits` produces bad exit code, so test fails.

[the remark this answers is not in the document]
Comment (yurikoles) [reply Formula/sdl2_sound.rb:151]: But I see both help and credits

[the remark this answers is not in the document]
Comment (SMillerDev) [reply Formula/sdl2_sound.rb:151]: Not to ignore, but you can pass the expected code as the second argument to `shell_output`

Comment (yurikoles) [inline Formula/sdl2_sound.rb:151]:
```suggestion
    system ENV.cc, pkgshare/"examples/playsound.c", "-o", "playsound", *flags
    Open3.popen3("#{bin}/playsound --help") do |_, _, stderr|
      assert_match "help", stderr.read
    end
```

Comment (pthariensflame) [reply Formula/sdl2_sound.rb:151]: Let's not overwrite the `playsound` that's built originally.

Comment (yurikoles) [reply Formula/sdl2_sound.rb:151]: I assume that you meant `playsound_simple`. The problem is that it either just plays a sound file or exits without much output, so I decided to drop it.

Comment (yurikoles) [reply Formula/sdl2_sound.rb:151]: Actually, they're different and `playsound_simple` even doesn't accept `--help`.

```
Couldn't load '--help': Couldn't open --help.
```

Comment (yurikoles) [inline Formula/sdl2_sound.rb:151]:
```suggestion
    system ENV.cc, pkgshare/"examples/playsound.c", "-o", "playsound", *flags
    assert_match "help", shell_output("./playsound --help 2>&1", 42)
```

Comment (yurikoles) [reply Formula/sdl2_sound.rb:151]: @pthariensflame you may ignore this suggestion.

Comment (yurikoles) [reply Formula/sdl2_sound.rb:151]: @pthariensflame please apply this one.

`playsound_simple` is basically cutted-out version of `playsound` that even don't accept something except sound file and prints nothing except error that there is no file.

**Review by SMillerDev — DISMISSED** [review]

**Review by chenrui333 — DISMISSED** [review]

File: Formula/sdl2_sound.rb [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+
+    flags = %W[
+      -I#{include}/SDL2
+      -I#{Formula["sdl2"].include}/SDL2
+      -L#{lib}
+      -L#{Formula["sdl2"].lib}
+      -lSDL2_sound
+      -lSDL2
+    ]
+    system ENV.cc, pkgshare/"examples/playsound.c", "-o", "playsound", *flags
```
Comment (yurikoles) [inline Formula/sdl2_sound.rb:149]:
```suggestion
    cp pkgshare/"examples/playsound.c", testpath
    
    if OS.Linux?
      inreplace "playsound.c",
        "#if HAVE_SIGNAL_H\n",
        "#define HAVE_SIGNAL_H 1\n#if HAVE_SIGNAL_H\n"
    end

    system ENV.cc, "playsound.c", "-o", "playsound", *flags
```

Comment (yurikoles) [reply Formula/sdl2_sound.rb:149]: @fxcoudert this patch fixes test

**Comment by yurikoles** [comment]
My suggestion was with trailing whitespace.

**Review by SMillerDev — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**